processor/tailsamplingprocessor/config.go

```diff
@@ -68,35 +69,23 @@ type SubPolicyCfg struct {
 	StringAttributeCfg StringAttributeCfg `mapstructure:"string_attribute"`
 	// Configs for rate limiting filter sampling policy evaluator.
 	RateLimitingCfg RateLimitingCfg `mapstructure:"rate_limiting"`
-	// Configs for and policy evaluator.
-	AndCfg AndCfg `mapstructure:"and"`
-	// Configs for span counter filter sampling policy evaluator.
+	// Configs for span count filter sampling policy evaluator.
 	SpanCountCfg SpanCountCfg `mapstructure:"span_count"`
-	// Configs for trace_state policy evaluator.
+	// Configs for defining trace_state policy
 	TraceStateCfg TraceStateCfg `mapstructure:"trace_state"`
 }
 
+// CompositeSubPolicyCfg holds the common configuration to all policies under composite policy.
+type CompositeSubPolicyCfg struct {
+	sharedPolicyCfg `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Configs for and policy evaluator.
+	AndCfg AndCfg `mapstructure:"and"`
+}
+
+// AndSubPolicyCfg holds the common configuration to all policies under and policy.
 type AndSubPolicyCfg struct {
```

Comment:
Do we need the `AndSubPolicyCfg` at all? if I understand correctly it's the same as `sharedPolicyCfg`